# 1. Import libraries

In [47]:
import pandas as pd
import numpy as np
import os

In [3]:
path = r'C:\Users\vikra\Instacart Basket Analysis 06_24'

In [4]:
df_merged = pd.read_csv( r'C:\Users\vikra\Instacart Basket Analysis 06_24\Data\Prepared Data\ords_prods_merge.csv', index_col = False)

In [22]:
 df_merged = df_merged.drop(columns = ['Unnamed: 0.1', 'Unnamed: 0', 'merge_indicator'])

In [5]:
df = df_merged[:1000000]


In [6]:
df.shape

(1000000, 17)

In [7]:
df.head()

Unnamed: 0.1  Unnamed: 0  product_id                product_name  aisle_id  \
0             0           0           1  Chocolate Sandwich Cookies        61   
1             1           0           1  Chocolate Sandwich Cookies        61   
2             2           0           1  Chocolate Sandwich Cookies        61   
3             3           0           1  Chocolate Sandwich Cookies        61   
4             4           0           1  Chocolate Sandwich Cookies        61   

   department_id  prices  order_id  add_to_cart_order  reordered  user_id  \
0             19     5.8      1107                  7          0    38259   
1             19     5.8      5319                  3          1   196224   
2             19     5.8      7540                  4          1   138499   
3             19     5.8      9228                  2          0    79603   
4             19     5.8      9273                 30          0    50005   

  eval_set  order_number  order_dow  order_hour_of_day  \
0    prior             2          1                 11   
1    prior            65          1                 14   
2    prior             8          0                 14   
3    prior             2          2                 10   
4    prior             1          1                 15   

   days_since_prior_order merge_indicator  
0                     7.0            both  
1                     1.0            both  
2                     7.0            both  
3                    30.0            both  
4                     NaN            both

In [8]:
df.groupby('product_name')


In [9]:
df.groupby('department_id').agg({'order_number': ['mean']})

order_number
                      mean
department_id             
1                15.562844
2                17.518895
3                16.889371
4                17.537934
5                14.894521
6                16.658449
7                17.072954
8                14.942021
9                15.447580
10               18.681852
11               15.331533
12               14.327957
13               16.578690
14               16.666369
15               16.153766
16               17.768070
17               15.602964
18               19.674252
19               16.924344
20               16.292631
21               25.535596

In [10]:
df.groupby('department_id')['order_number'].mean()

department_id
1     15.562844
2     17.518895
3     16.889371
4     17.537934
5     14.894521
6     16.658449
7     17.072954
8     14.942021
9     15.447580
10    18.681852
11    15.331533
12    14.327957
13    16.578690
14    16.666369
15    16.153766
16    17.768070
17    15.602964
18    19.674252
19    16.924344
20    16.292631
21    25.535596
Name: order_number, dtype: float64

In [11]:
df.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                15.562844   1  99
2                17.518895   1  96
3                16.889371   1  99
4                17.537934   1  99
5                14.894521   1  99
6                16.658449   1  99
7                17.072954   1  99
8                14.942021   1  98
9                15.447580   1  99
10               18.681852   1  99
11               15.331533   1  99
12               14.327957   1  99
13               16.578690   1  99
14               16.666369   1  99
15               16.153766   1  99
16               17.768070   1  99
17               15.602964   1  99
18               19.674252   1  99
19               16.924344   1  99
20               16.292631   1  99
21               25.535596   1  98

In [12]:
df.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                15.562844   1  99
2                17.518895   1  96
3                16.889371   1  99
4                17.537934   1  99
5                14.894521   1  99
6                16.658449   1  99
7                17.072954   1  99
8                14.942021   1  98
9                15.447580   1  99
10               18.681852   1  99
11               15.331533   1  99
12               14.327957   1  99
13               16.578690   1  99
14               16.666369   1  99
15               16.153766   1  99
16               17.768070   1  99
17               15.602964   1  99
18               19.674252   1  99
19               16.924344   1  99
20               16.292631   1  99
21               25.535596   1  98

# 2. To find the aggregated mean of the “order_number” column grouped by “department_id” for merged data.

In [43]:
df_merged.groupby('product_name')


In [44]:
df_merged.groupby('department_id').agg({'order_number': ['mean']})

order_number
                      mean
department_id             
1                15.457687
2                17.277920
3                17.179756
4                17.811403
5                15.215751
6                16.439806
7                17.225773
8                15.340520
9                15.895474
10               20.197148
11               16.170371
12               15.887622
13               16.583304
14               16.759763
15               16.165037
16               17.663250
17               15.694469
18               19.310397
19               17.177343
20               16.473447
21               22.902379

In [45]:
df_merged.groupby('department_id')['order_number'].mean()

department_id
1     15.457687
2     17.277920
3     17.179756
4     17.811403
5     15.215751
6     16.439806
7     17.225773
8     15.340520
9     15.895474
10    20.197148
11    16.170371
12    15.887622
13    16.583304
14    16.759763
15    16.165037
16    17.663250
17    15.694469
18    19.310397
19    17.177343
20    16.473447
21    22.902379
Name: order_number, dtype: float64

In [46]:
df_merged.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                15.457687   1  99
2                17.277920   1  99
3                17.179756   1  99
4                17.811403   1  99
5                15.215751   1  99
6                16.439806   1  99
7                17.225773   1  99
8                15.340520   1  99
9                15.895474   1  99
10               20.197148   1  99
11               16.170371   1  99
12               15.887622   1  99
13               16.583304   1  99
14               16.759763   1  99
15               16.165037   1  99
16               17.663250   1  99
17               15.694469   1  99
18               19.310397   1  99
19               17.177343   1  99
20               16.473447   1  99
21               22.902379   1  99

# 3. Analyze the result

While differing entire dataframe  from those of the subset
There are 21 depatment_ids  in both data set. Bur we could observe that , there is difference in the Mean Value.
for example: 
for department_id 1  Mean value is 15.457687 for Main dataframe.
for department_id 1  Mean value is 15.562844 for Main dataframe.

In [23]:
df_merged['max_order'] = df_merged.groupby(['user_id'])['order_number'].transform(np.max)


C:\Users\vikra\AppData\Local\Temp\ipykernel_3672\2077915841.py:1: FutureWarning: The provided callable <function max at 0x0000023D7C7E0720> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_merged['max_order'] = df_merged.groupby(['user_id'])['order_number'].transform(np.max)


In [24]:
df_merged.head(15)

product_id                product_name  aisle_id  department_id  prices  \
0            1  Chocolate Sandwich Cookies        61             19     5.8   
1            1  Chocolate Sandwich Cookies        61             19     5.8   
2            1  Chocolate Sandwich Cookies        61             19     5.8   
3            1  Chocolate Sandwich Cookies        61             19     5.8   
4            1  Chocolate Sandwich Cookies        61             19     5.8   
5            1  Chocolate Sandwich Cookies        61             19     5.8   
6            1  Chocolate Sandwich Cookies        61             19     5.8   
7            1  Chocolate Sandwich Cookies        61             19     5.8   
8            1  Chocolate Sandwich Cookies        61             19     5.8   
9            1  Chocolate Sandwich Cookies        61             19     5.8   
10           1  Chocolate Sandwich Cookies        61             19     5.8   
11           1  Chocolate Sandwich Cookies        61             19     5.8   
12           1  Chocolate Sandwich Cookies        61             19     5.8   
13           1  Chocolate Sandwich Cookies        61             19     5.8   
14           1  Chocolate Sandwich Cookies        61             19     5.8   

    order_id  add_to_cart_order  reordered  user_id eval_set  order_number  \
0       1107                  7          0    38259    prior             2   
1       5319                  3          1   196224    prior            65   
2       7540                  4          1   138499    prior             8   
3       9228                  2          0    79603    prior             2   
4       9273                 30          0    50005    prior             1   
5       9696                  5          1   108919    prior            46   
6      11140                  1          1    63782    prior             4   
7      11485                  4          0   170451    prior             5   
8      12672                  3          1   106854    prior            28   
9      13668                  4          1   181127    prior            10   
10     14668                 13          1   106519    prior            18   
11     16132                  1          1   145505    prior            64   
12     18303                  1          1   139923    prior            16   
13     19479                  5          0   110984    prior             2   
14     19569                  9          1   180719    prior            84   

    order_dow  order_hour_of_day  days_since_prior_order  max_order  
0           1                 11                     7.0          4  
1           1                 14                     1.0         99  
2           0                 14                     7.0         23  
3           2                 10                    30.0          5  
4           1                 15                     NaN         22  
5           5                 16                     8.0         51  
6           1                 14                    14.0          6  
7           5                 18                    24.0         10  
8           5                 10                     6.0         99  
9           4                 17                     9.0         22  
10          0                 22                     7.0         23  
11          2                 10                     5.0         67  
12          0                 23                    14.0         39  
13          2                 12                    20.0         16  
14          4                 19                     0.0         89

In [25]:
df_merged.shape

(32433030, 15)

In [28]:
df_merged.head(100)

product_id                product_name  aisle_id  department_id  prices  \
0            1  Chocolate Sandwich Cookies        61             19     5.8   
1            1  Chocolate Sandwich Cookies        61             19     5.8   
2            1  Chocolate Sandwich Cookies        61             19     5.8   
3            1  Chocolate Sandwich Cookies        61             19     5.8   
4            1  Chocolate Sandwich Cookies        61             19     5.8   
5            1  Chocolate Sandwich Cookies        61             19     5.8   
6            1  Chocolate Sandwich Cookies        61             19     5.8   
7            1  Chocolate Sandwich Cookies        61             19     5.8   
8            1  Chocolate Sandwich Cookies        61             19     5.8   
9            1  Chocolate Sandwich Cookies        61             19     5.8   
10           1  Chocolate Sandwich Cookies        61             19     5.8   
11           1  Chocolate Sandwich Cookies        61             19     5.8   
12           1  Chocolate Sandwich Cookies        61             19     5.8   
13           1  Chocolate Sandwich Cookies        61             19     5.8   
14           1  Chocolate Sandwich Cookies        61             19     5.8   
15           1  Chocolate Sandwich Cookies        61             19     5.8   
16           1  Chocolate Sandwich Cookies        61             19     5.8   
17           1  Chocolate Sandwich Cookies        61             19     5.8   
18           1  Chocolate Sandwich Cookies        61             19     5.8   
19           1  Chocolate Sandwich Cookies        61             19     5.8   
20           1  Chocolate Sandwich Cookies        61             19     5.8   
21           1  Chocolate Sandwich Cookies        61             19     5.8   
22           1  Chocolate Sandwich Cookies        61             19     5.8   
23           1  Chocolate Sandwich Cookies        61             19     5.8   
24           1  Chocolate Sandwich Cookies        61             19     5.8   
25           1  Chocolate Sandwich Cookies        61             19     5.8   
26           1  Chocolate Sandwich Cookies        61             19     5.8   
27           1  Chocolate Sandwich Cookies        61             19     5.8   
28           1  Chocolate Sandwich Cookies        61             19     5.8   
29           1  Chocolate Sandwich Cookies        61             19     5.8   
30           1  Chocolate Sandwich Cookies        61             19     5.8   
31           1  Chocolate Sandwich Cookies        61             19     5.8   
32           1  Chocolate Sandwich Cookies        61             19     5.8   
33           1  Chocolate Sandwich Cookies        61             19     5.8   
34           1  Chocolate Sandwich Cookies        61             19     5.8   
35           1  Chocolate Sandwich Cookies        61             19     5.8   
36           1  Chocolate Sandwich Cookies        61             19     5.8   
37           1  Chocolate Sandwich Cookies        61             19     5.8   
38           1  Chocolate Sandwich Cookies        61             19     5.8   
39           1  Chocolate Sandwich Cookies        61             19     5.8   
40           1  Chocolate Sandwich Cookies        61             19     5.8   
41           1  Chocolate Sandwich Cookies        61             19     5.8   
42           1  Chocolate Sandwich Cookies        61             19     5.8   
43           1  Chocolate Sandwich Cookies        61             19     5.8   
44           1  Chocolate Sandwich Cookies        61             19     5.8   
45           1  Chocolate Sandwich Cookies        61             19     5.8   
46           1  Chocolate Sandwich Cookies        61             19     5.8   
47           1  Chocolate Sandwich Cookies        61             19     5.8   
48           1  Chocolate Sandwich Cookies        61             19     5.8   
49           1  Chocolate Sandwich Cookies        61  

In [27]:
pd.options.display.max_rows = None


In [ ]:
# Check the regularity of customer 

In [30]:
df_merged.loc[df_merged['max_order'] > 40, 'loyalty_flag'] = 'Loyal customer'

In [34]:
df_merged.loc[(df_merged['max_order'] <= 40) & (df_merged['max_order'] > 10) ,'loyalty_flag'] = 'Regular customer'

In [32]:
df_merged.loc[df_merged['max_order'] < 10, 'loyalty_flag'] = 'New customer'

In [35]:
df_merged['loyalty_flag'].value_counts(dropna = False)

loyalty_flag
Regular customer    15890458
Loyal customer      10293434
New customer         5475359
NaN                   773779
Name: count, dtype: int64

In [38]:
df_merged.head()

product_id                product_name  aisle_id  department_id  prices  \
0           1  Chocolate Sandwich Cookies        61             19     5.8   
1           1  Chocolate Sandwich Cookies        61             19     5.8   
2           1  Chocolate Sandwich Cookies        61             19     5.8   
3           1  Chocolate Sandwich Cookies        61             19     5.8   
4           1  Chocolate Sandwich Cookies        61             19     5.8   

   order_id  add_to_cart_order  reordered  user_id eval_set  order_number  \
0      1107                  7          0    38259    prior             2   
1      5319                  3          1   196224    prior            65   
2      7540                  4          1   138499    prior             8   
3      9228                  2          0    79603    prior             2   
4      9273                 30          0    50005    prior             1   

   order_dow  order_hour_of_day  days_since_prior_order  max_order  \
0          1                 11                     7.0          4   
1          1                 14                     1.0         99   
2          0                 14                     7.0         23   
3          2                 10                    30.0          5   
4          1                 15                     NaN         22   

       loyalty_flag  
0      New customer  
1    Loyal customer  
2  Regular customer  
3      New customer  
4  Regular customer

In [40]:
df_merged [['user_id', 'loyalty_flag', 'order_number']].head()

user_id      loyalty_flag  order_number
0    38259      New customer             2
1   196224    Loyal customer            65
2   138499  Regular customer             8
3    79603      New customer             2
4    50005  Regular customer             1

In [ ]:
New customers have the least price of products in both the average and max. 
Loyal customer have the highest price of products in both the average and max."
   ]


# 4. & 5. Creating column for average spender


In [48]:
df_merged['average_spend'] = df_merged.groupby(['user_id'])['prices'].transform(np.mean)

C:\Users\vikra\AppData\Local\Temp\ipykernel_3672\2240020810.py:1: FutureWarning: The provided callable <function mean at 0x0000023D7C7E1120> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  df_merged['average_spend'] = df_merged.groupby(['user_id'])['prices'].transform(np.mean)


In [51]:
df_merged [['user_id' ,'order_number' ,'loyalty_flag', 'average_spend']].head()

user_id  order_number      loyalty_flag  average_spend
0    38259             2      New customer       7.163333
1   196224            65    Loyal customer       5.342729
2   138499             8  Regular customer       7.469540
3    79603             2      New customer       6.986207
4    50005             1  Regular customer       7.493154

 # 6. To check different types of spenders like Low spender & High spender.

In [52]:
df_merged.loc[df_merged['average_spend'] < 10, 'spender_flag'] = 'Low_spender'

In [53]:
df_merged.loc[df_merged['average_spend'] >= 10, 'spender_flag'] = 'High_spender'

In [54]:
df_merged['spender_flag'].value_counts(dropna = False)

spender_flag
Low_spender     31797576
High_spender      635454
Name: count, dtype: int64

In [61]:
df_merged [['user_id' ,'order_number' ,'loyalty_flag', 'average_spend', 'spender_flag', 'prices']].head(100)

user_id  order_number      loyalty_flag  average_spend spender_flag  \
0     38259             2      New customer       7.163333  Low_spender   
1    196224            65    Loyal customer       5.342729  Low_spender   
2    138499             8  Regular customer       7.469540  Low_spender   
3     79603             2      New customer       6.986207  Low_spender   
4     50005             1  Regular customer       7.493154  Low_spender   
5    108919            46    Loyal customer       7.092840  Low_spender   
6     63782             4      New customer       5.700000  Low_spender   
7    170451             5               NaN       6.534286  Low_spender   
8    106854            28    Loyal customer       8.594301  Low_spender   
9    181127            10  Regular customer       7.177143  Low_spender   
10   106519            18  Regular customer       6.582996  Low_spender   
11   145505            64    Loyal customer       7.179795  Low_spender   
12   139923            16  Regular customer       6.353535  Low_spender   
13   110984             2  Regular customer       9.758140  Low_spender   
14   180719            84    Loyal customer       8.619501  Low_spender   
15   142388             7  Regular customer       6.654412  Low_spender   
16   200356            11  Regular customer       8.658140  Low_spender   
17     4122            11  Regular customer       7.598765  Low_spender   
18   162394             2      New customer       7.355556  Low_spender   
19   148255            18  Regular customer       5.983893  Low_spender   
20   205157            11  Regular customer       6.322222  Low_spender   
21   181127             3  Regular customer       7.177143  Low_spender   
22    92648             8  Regular customer       6.060989  Low_spender   
23    30345             6      New customer       5.844000  Low_spender   
24    50898            28  Regular customer       7.315333  Low_spender   
25    57996            12    Loyal customer       6.517469  Low_spender   
26    77956            90    Loyal customer       7.289769  Low_spender   
27    22561             4      New customer       6.823810  Low_spender   
28   199115            20  Regular customer       6.740860  Low_spender   
29    91160            32    Loyal customer       7.317925  Low_spender   
30    50150            17  Regular customer       7.997849  Low_spender   
31    38748            16  Regular customer       7.955443  Low_spender   
32    65777            47    Loyal customer       6.275781  Low_spender   
33   134728            18  Regular customer       6.153191  Low_spender   
34    11119            18  Regular customer       5.878641  Low_spender   
35    45784             4  Regular customer       6.230976  Low_spender   
36    21054             3      New customer       5.189655  Low_spender   
37   196224            23    Loyal customer       5.342729  Low_spender   
38   102960            21    Loyal customer       7.327751  Low_spender   
39   113429            17  Regular customer       7.479439  Low_spender   
40   111982            76    Loyal customer       4.827700  Low_spender   
41   144159             3  Regular customer       6.214487  Low_spender   
42    72224            12    Loyal customer       6.765598  Low_spender   
43    23234            10  Regular customer       5.266667  Low_spender   
44   131925             8  Regular customer       7.351899  Low_spender   
45    16057             2  Regular customer       6.862338  Low_spender   
46   124716             5      New customer       6.517857  Low_spender   
47    60380            26  Regular customer       7.063025  Low_spender   
48   197324            12  Regular customer       8.303311  Low_spender   
49   157800             9  Regular customer       7.822115  Low_spender   
50    48228             1      New customer       5.998630  Low_spender   
51    23513             5  Regular customer       6.842857  Low_spender   
52    95847            44    

# 7.To determine frequent versus non-frequent customers

In [63]:
# creating column 
df_merged['customer_frequency'] = df_merged.groupby(['user_id'])['days_since_prior_order'].transform(np.median)

C:\Users\vikra\AppData\Local\Temp\ipykernel_3672\400500674.py:2: FutureWarning: The provided callable <function median at 0x0000023D7C903600> is currently using SeriesGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  df_merged['customer_frequency'] = df_merged.groupby(['user_id'])['days_since_prior_order'].transform(np.median)


# Creating flag 

In [64]:
df_merged.loc[df_merged['customer_frequency'] > 20, 'frequency_flag'] = 'Non-frequent customer'

In [65]:
df_merged.loc[(df_merged['customer_frequency'] > 10 ) & (df_merged['customer_frequency'] <= 20) ,'loyalty_flag'] = 'Regular customer'

In [67]:
df_merged.loc[df_merged['customer_frequency'] <= 10, 'frequency_flag'] = 'Frequent customer'

In [68]:
df_merged['customer_frequency'].value_counts(dropna = False)

customer_frequency
7.0     5688169
6.0     2985726
8.0     2714922
5.0     2491723
4.0     2315715
30.0    1684509
9.0     1638067
3.0     1482143
10.0    1273145
14.0    1214572
13.0    1058938
11.0    1051237
12.0     974131
15.0     744511
2.0      561600
16.0     515950
21.0     421949
17.0     399638
20.0     371611
18.0     357944
19.0     349671
22.0     266181
23.0     202481
28.0     200151
27.0     173625
24.0     169169
26.0     162200
25.0     152535
1.0      151844
29.0     137177
7.5       49560
6.5       43735
8.5       37149
0.0       34237
9.5       31409
10.5      30334
5.5       29117
4.5       28269
12.5      25556
13.5      23105
11.5      19808
14.5      18494
15.5      14682
18.5      14539
17.5      12759
3.5       11762
20.5       9860
16.5       9722
19.5       9442
25.5       7724
22.5       7680
2.5        7354
24.5       7323
21.5       7093
28.5       6781
23.5       6777
27.5       5710
26.5       5481
29.5       5017
1.5        1058
0.5         259
Name:

In [71]:
df_merged [['user_id' ,'order_number' ,'loyalty_flag', 'average_spend', 'spender_flag', 'prices','customer_frequency']].head(100)

user_id  order_number      loyalty_flag  average_spend spender_flag  \
0     38259             2      New customer       7.163333  Low_spender   
1    196224            65    Loyal customer       5.342729  Low_spender   
2    138499             8  Regular customer       7.469540  Low_spender   
3     79603             2      New customer       6.986207  Low_spender   
4     50005             1  Regular customer       7.493154  Low_spender   
5    108919            46    Loyal customer       7.092840  Low_spender   
6     63782             4  Regular customer       5.700000  Low_spender   
7    170451             5  Regular customer       6.534286  Low_spender   
8    106854            28    Loyal customer       8.594301  Low_spender   
9    181127            10  Regular customer       7.177143  Low_spender   
10   106519            18  Regular customer       6.582996  Low_spender   
11   145505            64    Loyal customer       7.179795  Low_spender   
12   139923            16  Regular customer       6.353535  Low_spender   
13   110984             2  Regular customer       9.758140  Low_spender   
14   180719            84    Loyal customer       8.619501  Low_spender   
15   142388             7  Regular customer       6.654412  Low_spender   
16   200356            11  Regular customer       8.658140  Low_spender   
17     4122            11  Regular customer       7.598765  Low_spender   
18   162394             2      New customer       7.355556  Low_spender   
19   148255            18  Regular customer       5.983893  Low_spender   
20   205157            11  Regular customer       6.322222  Low_spender   
21   181127             3  Regular customer       7.177143  Low_spender   
22    92648             8  Regular customer       6.060989  Low_spender   
23    30345             6      New customer       5.844000  Low_spender   
24    50898            28  Regular customer       7.315333  Low_spender   
25    57996            12    Loyal customer       6.517469  Low_spender   
26    77956            90    Loyal customer       7.289769  Low_spender   
27    22561             4  Regular customer       6.823810  Low_spender   
28   199115            20  Regular customer       6.740860  Low_spender   
29    91160            32    Loyal customer       7.317925  Low_spender   
30    50150            17  Regular customer       7.997849  Low_spender   
31    38748            16  Regular customer       7.955443  Low_spender   
32    65777            47    Loyal customer       6.275781  Low_spender   
33   134728            18  Regular customer       6.153191  Low_spender   
34    11119            18  Regular customer       5.878641  Low_spender   
35    45784             4  Regular customer       6.230976  Low_spender   
36    21054             3      New customer       5.189655  Low_spender   
37   196224            23    Loyal customer       5.342729  Low_spender   
38   102960            21    Loyal customer       7.327751  Low_spender   
39   113429            17  Regular customer       7.479439  Low_spender   
40   111982            76    Loyal customer       4.827700  Low_spender   
41   144159             3  Regular customer       6.214487  Low_spender   
42    72224            12    Loyal customer       6.765598  Low_spender   
43    23234            10  Regular customer       5.266667  Low_spender   
44   131925             8  Regular customer       7.351899  Low_spender   
45    16057             2  Regular customer       6.862338  Low_spender   
46   124716             5      New customer       6.517857  Low_spender   
47    60380            26  Regular customer       7.063025  Low_spender   
48   197324            12  Regular customer       8.303311  Low_spender   
49   157800             9  Regular customer       7.822115  Low_spender   
50    48228             1      New customer       5.998630  Low_spender   
51    23513             5  Regular customer       6.842857  Low_spender   
52    95847            44    

# 8. & 9. Export your dataframe as a pickle file

In [72]:
df_merged.to_pickle(os.path.join(path, 'Data', 'Prepared Data', 'ords_prods_merged_grouped.pkl'))